src/pkg/services/m365/api/lists.go

```diff
@@ -547,9 +561,20 @@ func populateMultipleValues(val any, filteredData map[string]any, colDetails *co
 
 	lookupIDs := make([]float64, 0)
 
+	checkFields := func(colDetails *columnDetails) []string {
+		if colDetails.isLookupColumn {
```

Comment:
a switch statement might be better here.

Also, why an ad-hoc func?  There's nothing here dependent upon runtime state.  Seems like you can move it to an official func and give it some quick unit tests.